akka-javasdk-tests/src/test/java/akkajavasdk/components/agent/TestModelProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akkajavasdk.components.agent;
+
+import akka.japi.Pair;
+import akka.javasdk.agent.ModelProvider;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.TextContent;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.chat.ChatModel;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.output.FinishReason;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+public class TestModelProvider implements ModelProvider.Custom {
+
+  private List<Pair<Predicate<String>, String>> responsePredicates = new ArrayList<>();
+
+  class TestChatResponse extends ChatResponse {
```

Comment:
Why do we need the subclass TestChatResponse?
You can create it with 
```
ChatResponse.builder()
  .modelName("test-model")
  .finishReason(FinishReason.STOP)
  .aiMessage(AiMessage.from(response))
```

Reply:
true, removed